distribution/tools/plugin-cli/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -555,25 +582,26 @@ private PluginInfo verify(Terminal terminal, Path pluginRoot, boolean isBatch, E
         }
 
         // check for jar hell before any copying
-        jarHellCheck(pluginRoot, env.pluginsFile());
+        jarHellCheck(info, pluginRoot, env.pluginsFile());
 
         // read optional security policy (extra permissions)
         // if it exists, confirm or warn the user
         Path policy = pluginRoot.resolve(PluginInfo.ES_PLUGIN_POLICY);
         if (Files.exists(policy)) {
             PluginSecurity.readPolicy(info, policy, terminal, env::tmpFile, isBatch);
         }
-
-        return info;
     }
 
     /** check a candidate plugin for jar hell before installing it */
-    void jarHellCheck(Path candidate, Path pluginsDir) throws Exception {
+    void jarHellCheck(PluginInfo candidateInfo, Path candidate, Path pluginsDir) throws Exception {
```

Comment:
nit: can you please rename `candidate` to `candidateDir` to be parallel with `candidateInfo`?